lib/client/bst-config.ts

```diff
@@ -131,19 +132,61 @@ export class BSTConfig {
         if (isUUIDUnassigned) {
             return spokesClient.createPipe();
         }
-        throw Error("Unable to create spokes connection");
+
+        // Pipe exists, we return the info we have as pipe
+        return {
+            endPoint: {
+                name: id,
+            },
+            uuid: secretKey,
+        };
     }
 
-    private static async createExternalResources(nodeID?: string): Promise<any> {
+    private static async createSource(secretKey?: string, sourceID?: string): Promise<any> {
         const sourceNameGenerator = new SourceNameGenerator();
-        const generatedKey = await sourceNameGenerator.callService();
-        // This is for backwards compatibility - we use the nodeID for the secretKey if there is already a node-id.
-        // That way, the user does not need to change their configuration
-        const secretKey = nodeID ? nodeID : generatedKey.secretKey;
-        const requests = [this.createSpokesPipe(generatedKey.id, secretKey),
-                            sourceNameGenerator.createDashboardSource(generatedKey.id, secretKey)];
-        const responses = await Promise.all(requests);
-        return responses[0];
+        let id;
+        let key;
+
+        // Covers new users case and also
+        // If someone have only sourceID but not secretKey then he modified the config, so it's ok to drop
+        if (!secretKey) {
+            const generatedKey = await sourceNameGenerator.callService();
+            id = generatedKey.id;
+            key = generatedKey.key;
+        }
+
+        // This means it has nodeID but have no pipe or key
+        if (secretKey && !sourceID) {
+            const generatedKey = await sourceNameGenerator.callService();
+            id = generatedKey.id;
+            key = secretKey;
+        }
+
+        if (sourceID && secretKey) {
+            // We have a previously created config, we try to create it directly
+            id = sourceID;
+            key = secretKey;
+        }
+
+        try {
+            await sourceNameGenerator.createDashboardSource(id, key);
+        } catch (e) {
+            // If the source already exists everything is fine
+            if (e.statusCode !== 403) {
```

Comment:
When the source exists the source generator error is:
403 The source already exists (checked that on postman)
We want to be able to still send an error in other cases. I wanted to restrict to the statusCode plus the message , but when testing it locally it did send the 403 but the error message was unknown